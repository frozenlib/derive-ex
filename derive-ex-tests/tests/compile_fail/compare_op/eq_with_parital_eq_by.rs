use derive_ex::derive_ex;

#[derive_ex(Eq, PartialEq)]
struct X(#[partial_eq(by = |l, r| l.len() == r.len())] String);

fn main() {}
