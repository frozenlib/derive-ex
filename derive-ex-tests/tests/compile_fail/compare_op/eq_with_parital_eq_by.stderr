error: When `#[derive_ex(Eq)]` is specified, `#[eq(by = ...)]` must be used instead of `#[partial_eq(by = ...)]`.
 --> tests/compile_fail/compare_op/eq_with_parital_eq_by.rs:4:28
  |
4 | struct X(#[partial_eq(by = |l, r| l.len() == r.len())] String);
  |                            ^
